tests/baselines/reference/pathMappingBasedModuleResolution1_amd.errors.txt

```diff
@@ -0,0 +1,28 @@
+error TS-1: Pre-emit (1) and post-emit (2) diagnostic counts do not match! This can indicate that a semantic _error_ was added by the emit resolver - such an error may not be reflected on the command line or in the editor, but may be captured in a baseline here!
+c:/root/tsconfig.json(5,17): error TS5089: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?
+c:/root/tsconfig.json(6,17): error TS5089: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?
+
+
+!!! error TS-1: Pre-emit (1) and post-emit (2) diagnostic counts do not match! This can indicate that a semantic _error_ was added by the emit resolver - such an error may not be reflected on the command line or in the editor, but may be captured in a baseline here!

```

Comment:
@weswigham this showed up when I rebased on master—is it a problem, and if so, how do I get rid of it?

Reply:
> is it a problem,

Yep! It means the diagnostics being reported are dependent on weather emit happens before or after diagnostics are requested! (Which is bad) You should endeavor to prevent this from occurring - usually that means checking for an error from all possible entrypoints for that error, rather than just one (eg, an issue we had in the near past was we only reported a duplicate error on the "first" declaration of a symbol - if they were in different files, this meant the error could go unreported if emit occurred before a full check).

Reply:
Essentially, the gist of the error is that if you ran this test on the command line, rather than in the test harness, those two errors would never be reported (because they got added during the transform or emit phase, after diagnostics were reported).

Reply:
I have no idea how this could be true, because it’s just a config file diagnostic generated during program construction like every other config file diagnostic: https://github.com/microsoft/TypeScript/pull/40101/files#diff-08a3cc4f1f9a51dbb468c2810f5229d3R3048

Reply:
Well... You can look into the test harness for how it checks this - it makes a program and requests diagnostics, then makes a program, emits, and requests diagnostics, and compares the two results. Do we have other tests that check config diagnostics in this way? If not, you might be be the first to uncover the issue.

Reply:
It was a problem with the harness spreading away non-enumerable properties from `compilerOptions`. Fixed 👍